### DISCUSSION TITLE
How To Upload To Cloudinary?

### DISCUSSION THREAD

**Discussion by ismailbentabett** [awcodes/filament-curator#306] [Q&A]
i tried everything to make the upload to cloudinary work but i keep getting resource not found 

CustomServerFactory.php :
```
 return ServerFactory::create([
            'driver' => 'imagick',
            'response' => new LaravelResponseFactory(app('request')),
            'source' => Storage::disk('cloudinary')->getDriver(),
            'source_path_prefix' => 'media',
            'cache' => storage_path('app'),
            'cache_path_prefix' => '.cache',
            'max_image_size' => 2000 * 2000,

        ]);
```
curator.php :
```
return [
    'accepted_file_types' => [
        'image/jpeg',
        'image/png',
        'image/webp',
        'image/svg+xml',
        'application/pdf',
        //videos
        'video/mp4',
        'video/ogg',
        'video/webm',
        //audio
        'audio/mpeg',
        'audio/ogg',
        'audio/wav',
        'audio/webm',
        //docs
        'application/msword',
        'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
        'application/vnd.ms-excel',
        'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
        'application/vnd.ms-powerpoint',
        'application/vnd.openxmlformats-officedocument.presentationml.presentation',

    ],
    'cloud_disks' => [
        's3',
        'cloudinary',
        'imgix',
    ],
    'curation_presets' => [
        \Awcodes\Curator\Curations\ThumbnailPreset::class,
    ],
    'directory' => 'media',
    'disk' => env('FILAMENT_FILESYSTEM_DISK', 'cloudinary'),
    'glide' => [
        'server' => CustomServerFactory::class,

    ],
    'image_crop_aspect_ratio' => null,
    'image_resize_mode' => null,
    'image_resize_target_height' => null,
    'image_resize_target_width' => null,
    'is_limited_to_directory' => false,
    'max_size' => 10000,
    'model' => \Awcodes\Curator\Models\Media::class,
    'min_size' => 0,
    'path_generator' => null,
    'resources' => [
        'label' => 'Media',
        'plural_label' => 'Media',
        'navigation_group' => null,
        'navigation_icon' => 'heroicon-o-photo',
        'navigation_sort' => null,
        'navigation_count_badge' => false,
        'resource' => \Awcodes\Curator\Resources\MediaResource::class,
    ],
    'should_preserve_filenames' => true,
    'should_register_navigation' => true,
    'visibility' => 'public',
];
```
fileSystem.php : 
```
 'cloudinary' => [
            'driver' => 'cloudinary',
            'cloud_name' => env('CLOUDINARY_CLOUD_NAME'),
            'api_key' => env('CLOUDINARY_API_KEY'),
            'api_secret' => env('CLOUDINARY_API_SECRET'),
            'url' => [
                'secure' => (bool) env('CLOUDINARY_SECURE_URL', true),
            ],
```

i keep getting : 

![image](https://github.com/awcodes/filament-curator/assets/42011079/d5e0f463-b6a9-4646-b572-93f67ce2fe0f)

**Comment by ismailbentabett** [comment]
pdf files and videos are uploading successfully

**Comment by awcodes** [comment]
What flysystem adapter are you using?

**Reply by ismailbentabett** [reply]
i am using this https://github.com/codebar-ag/laravel-flysystem-cloudinary

**Comment by awcodes** [comment]
Hmm. What's happening is that, with images, etc, Intervention is trying to create a new instance of Image so that it can read the width, height, etc, to store that information in the database. But it seems like it can't read the tmp file that is uploaded. Typically, I see this issue with s3 when the bucket permissions aren't configured correctly.

I have an upcoming project that needs Cloudinary so I can do some more digging then. But would love to hear your feedback if you can do some debugging.

**Comment by ismailbentabett** [comment]
I Really Hoped For a Solution But i found nothing

**Comment by awcodes** [comment]
Are your temp files getting uploaded to Cloudinary or to livewire-tmp on your server?

**Reply by ismailbentabett** [reply]
temporary files g reside on  server and are not automatically uploaded to Cloudinary

**Reply by awcodes** [reply]
I'm surprised your getting an error then. Is there a stack trace or flair url you can share?

**Comment by awcodes** [comment]
@ismailbentabett 

Hey, looking into this today, and I think that this [https://github.com/codebar-ag/laravel-flysystem-cloudinary#-file-extension-problem](https://github.com/codebar-ag/laravel-flysystem-cloudinary#-file-extension-problem) could be an issue with this particular driver and Curator.

**Comment by awcodes** [comment]
@ismailbentabett 

Think I got you sorted with this. It's working for me with the [Cloudinary Laravel SDK](https://github.com/cloudinary-devs/cloudinary-laravel)

You will need to use a custom model though to bypass Glide so the images are delivered from Cloudinary. Here's a gist with an example.

[https://gist.github.com/awcodes/4cd666ec1b984dd1ef119e7f7ea4066d](https://gist.github.com/awcodes/4cd666ec1b984dd1ef119e7f7ea4066d)

I still think that particular driver is going to give you issues though. :)

Cheers.
